### ISSUE TITLE
AttributeError while running bench_gpu_1bn.py

### ISSUE THREAD

**Issue by msharmavikram** [facebookresearch/faiss#543]
# Summary

I am trying to run bench_gpu_1bn.py with bigann dataset with GPU and get AttributeError: type object 'object' has no attribute '__getattr__'

Initially, I suspected the issue was from installation and hence re-installed it and experimented with bench_gpu_sift1m.py. This ran smoothly. I am not sure what is wrong with the bigann. 

# Platform

OS:  Ubuntu 16.04

Faiss version:  2dc30e14cfa76985ff8c2821c051f1725fe66afa

Faiss compilation options:  makefile.inc 

```
# Copyright (c) 2015-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed under the BSD+Patents license found in the
# LICENSE file in the root directory of this source tree.

CXX          = g++ -std=c++11
CXXCPP       = g++ -std=c++11 -E
# TODO: Investigate the LAPACKE wrapper for LAPACK, which defines the correct
#   type for FORTRAN integers.
CPPFLAGS     = -DFINTEGER=int 
CXXFLAGS     = -fPIC -fopenmp -m64 -Wno-sign-compare -g -O3 -Wall -Wextra
CPUFLAGS     = -msse4 -mpopcnt
LDFLAGS      = -fopenmp 
LIBS         = -lopenblas  
PYTHONCFLAGS = -I/usr/include/python2.7 -I/usr/include/python2.7 -I/usr/lib64/python2.7/site-packages/numpy/core/include

NVCC         = /usr/local/cuda/bin/nvcc
NVCCLDFLAGS  = -L/usr/local/cuda/lib64 
NVCCLIBS     = -lcudart -lcublas -lcuda 
CUDAROOT     = /usr/local/cuda
CUDACFLAGS   = -I/usr/local/cuda/include 
NVCCFLAGS    = -I $(CUDAROOT)/targets/x86_64-linux/include/ \
-Xcompiler -fPIC \
-Xcudafe --diag_suppress=unrecognized_attribute \
-gencode arch=compute_35,code="compute_35" \
-gencode arch=compute_52,code="compute_52" \
-gencode arch=compute_60,code="compute_60" \
-gencode arch=compute_61,code="compute_61" \
-gencode arch=compute_70,code="compute_70" \
-lineinfo \
-ccbin $(CXX) -DFAISS_USE_FLOAT16

OS = $(shell uname -s)

SHAREDEXT   = so
SHAREDFLAGS = -shared

ifeq ($(OS),Darwin)
	SHAREDEXT   = dylib
	SHAREDFLAGS = -dynamiclib -undefined dynamic_lookup
endif

MKDIR_P      = /usr/bin/mkdir -p
PYTHON       = python
SWIG         = swig

prefix      ?= /usr/local
exec_prefix ?= ${prefix}
libdir       = ${exec_prefix}/lib
includedir   = ${prefix}/include

```
 
Running on:
- [X] CPU
- [X] GPU

Interface: 
- [ ] C++
- [X] Python

# Reproduction instructions:
(I dont think IVF32768 matters. Was able to repeat the same issue with IVK256K)
```
python bench_gpu_1bn.py SIFT1M OPQ8_32,IVF32768,PQ8 -nnn 10 -ngpu 1 -tempmem $[1024*1024*1024*12] 

Preparing dataset SIFT1M
sizes: B (1000000, 128) Q (10000, 128) T (100000000, 128) gt (10000, 1000)
cachefiles:
/tmp/bench_gpu_1bn/preproc_SIFT1M_OPQ8_32.vectrans
/tmp/bench_gpu_1bn/cent_SIFT1M_OPQ8_32,IVF32768.npy
/tmp/bench_gpu_1bn/SIFT1M_OPQ8_32,IVF32768,PQ8.index
preparing resources for 1 GPUs
train preproc OPQ8_32,
preproc train done in 76.499 s
store /tmp/bench_gpu_1bn/preproc_SIFT1M_OPQ8_32.vectrans
train coarse quantizer...
apply preproc on shape (8388608, 128) k= 32768
   preproc 5.505 s output shape (8388608, 32)
Clustering 8388608 points in 32D to 32768 clusters, redo 1 times, 25 iterations
  Preprocessing in 2.42 s
  Iteration 24 (103.73 s, search 96.33 s): objective=2.15484e+11 imbalance=1.121 nsplit=0       
Coarse train time: 112.532 s
store centroids /tmp/bench_gpu_1bn/cent_SIFT1M_OPQ8_32,IVF32768.npy
making an IVFPQ index, m =  8
Training vector codes
  done 16.518 s
  IndexIVFPQ size 0 -> GpuIndexIVFPQ indicesOptions=0 usePrecomputed=1 useFloat16=0 reserveVecs=1000000
add...
983040/1000000 (2.964 s)   Add time: 2.964 s
Aggregate indexes to CPU
Traceback (most recent call last):
  File "bench_gpu_1bn.py", line 744, in <module>
    index = get_populated_index(preproc)
  File "bench_gpu_1bn.py", line 616, in get_populated_index
    gpu_index, indexall = compute_populated_index(preproc)
  File "bench_gpu_1bn.py", line 555, in compute_populated_index
    index_src = faiss.index_gpu_to_cpu(gpu_index.at(i))
  File "/home/vsm2/deepstore/vsm2_faiss/python/faiss/swigfaiss_gpu.py", line 5018, in <lambda>
    __getattr__ = lambda self, name: _swig_getattr(self, GpuIndexIVFPQ, name)
  File "/home/vsm2/deepstore/vsm2_faiss/python/faiss/swigfaiss_gpu.py", line 74, in _swig_getattr
    return _swig_getattr_nondynamic(self, class_type, name, 0)
  File "/home/vsm2/deepstore/vsm2_faiss/python/faiss/swigfaiss_gpu.py", line 69, in _swig_getattr_nondynamic
    return object.__getattr__(self, name)
AttributeError: type object 'object' has no attribute '__getattr__'

```

**Comment by beauby** [comment]
What are your python/swig versions?

**Comment by msharmavikram** [comment]
```
Python : 2.7.12
swig : 3.0.8
```

**Comment by mdouze** [comment]
I think the script has not been tested with a single GPU. It probably needs to be adapted a bit. Feel free to submit a PR to fix.

**Comment by mdouze** [comment]
Fixed in 1.4.0
